### PR TITLE
[Feature #103] Tabular module - change method responsibility

### PR DESCRIPTION
**Pull request by Matthew-Kulich**

### CODE REVIEW

File: s-pipes-modules/module-tabular/src/main/java/cz/cvut/spipes/modules/model/Column.java [hunk 1]
```diff
@@ -120,33 +127,52 @@ public String getProperty() {
...
-     * @param variableSetter The setter of the column variable
-     * @param variableName The name of the column variable we want to set
-     */
-    private void setColumnVariable(String schemaBasedValue, String dataBasedValue,
-                                   UnaryOperator<String> variableSetter, String variableName) {
-        if (schemaBasedValue != null){
-            checkVariable(schemaBasedValue, dataBasedValue, variableName);
-        }else {
-            variableSetter.apply(dataBasedValue);
+    public void setProperty(String dataPrefix, StreamResource sourceResource) throws UnsupportedEncodingException {
```
Comment (blcham) [inline s-pipes-modules/module-tabular/src/main/java/cz/cvut/spipes/modules/model/Column.java:130]: Agreed, we do not want to depend on StreamResource ... I would put url.

File: s-pipes-modules/module-tabular/src/main/java/cz/cvut/spipes/modules/model/Column.java [hunk 2]
```diff
@@ -120,33 +127,52 @@ public String getProperty() {
...
+        String propertyValue = getPropertyUrl(dataPrefix, sourceResource);
+        tabularModuleUtils.setVariable(this.property, propertyValue, value -> this.property = value, "property");
+        tabularModuleUtils.setVariable(this.propertyUrl, propertyValue, value -> this.propertyUrl = value, "propertyUrl");
+    }
+
+    private String getPropertyUrl(String dataPrefix, StreamResource sourceResource)
```
Comment (blcham) [inline s-pipes-modules/module-tabular/src/main/java/cz/cvut/spipes/modules/model/Column.java:136]: same as above.

File: s-pipes-modules/module-tabular/src/main/java/cz/cvut/spipes/modules/model/Column.java [hunk 3]
```diff
@@ -120,33 +127,52 @@ public String getProperty() {
...
+        if(getValueUrl() != null) valueURL = getValueUrl();
+
+        if (valueURL != null && !valueURL.isEmpty()) {
+            setValueUrl(valueURL);
+        } else {
+            if (cellValue != null) {
+                Resource columnResource = ResourceFactory
+                        .createResource(createAboutUrl(tableSchemaAboutUrl, rowNumber));
+
+                outputModel.add(ResourceFactory.createStatement(
```
Comment (blcham) [inline s-pipes-modules/module-tabular/src/main/java/cz/cvut/spipes/modules/model/Column.java:161]: this is very strange ... why column modifies output model ?

File: s-pipes-modules/module-tabular/src/main/java/cz/cvut/spipes/modules/model/TableSchema.java [hunk 4]
```diff
@@ -63,12 +81,91 @@ public List<Column> sortColumns(List<String> orderList){
...
-            Optional<Column> col = columnsSet.stream().filter(column -> column.getUri().toString().equals(uri)).findFirst();
-
+            Optional<Column> col = columnsSet.stream()
+                    .filter(column -> column.getUri().toString().equals(uri))
+                    .findFirst();
             col.ifPresent(columnList::add);
         }
         return columnList;
     }
+
```
Comment (blcham) [inline s-pipes-modules/module-tabular/src/main/java/cz/cvut/spipes/modules/model/TableSchema.java:92]: same as above.

File: s-pipes-modules/module-tabular/src/main/java/cz/cvut/spipes/modules/model/Column.java (hunk 3 above)
Comment (blcham) [reply s-pipes-modules/module-tabular/src/main/java/cz/cvut/spipes/modules/model/Column.java:161]: - issues of the function:
  - it is not very clear what method does -- it does at least 2 things
  - it modifies outputModel which is outside of the column ... so multiple objects modifies output model which might be quite dangerous ... you might not be able to track easily when outputModel was modified during the transformation -- this should be quite clear not spread in different objects 
   - not sure if  "createAboutUrl(tableSchemaAboutUrl, rowNumber)" is not responsibility of tableSchema (maybe it is all right ... just consider it)

Comment (blcham) [reply s-pipes-modules/module-tabular/src/main/java/cz/cvut/spipes/modules/model/Column.java:161]: btw. i stopped seeing references to the specification (i hope we did not loose it :) :) :)

**Review by blcham — CHANGES_REQUESTED** [review]
see my comments

File: s-pipes-modules/module-tabular/src/main/java/cz/cvut/spipes/modules/model/Column.java (hunk 3 above)
Comment (Matthew-Kulich) [reply s-pipes-modules/module-tabular/src/main/java/cz/cvut/spipes/modules/model/Column.java]: We didn't lose it :)) It is still there.

**Review by blcham — APPROVED** [review]

**Pull request merged**